cobalt/worker/service_worker_registration_map.cc

```diff
@@ -70,12 +70,8 @@ void ServiceWorkerRegistrationMap::ReadPersistentSettings() {
 }
 
 void ServiceWorkerRegistrationMap::DeletePersistentSettings() {
-  base::OnceClosure closure = base::BindOnce(
-      [](std::map<RegistrationMapKey,
-                  scoped_refptr<ServiceWorkerRegistrationObject>>*
-             registration_map) { (*registration_map).clear(); },
```

Comment:
Ohhh. How this was implemented this would call this `clear()` from the `PersistentSettings` thread without locking the map (meaning crashes if the map is accessed at the same time from the `ServiceWorker` thread)...  When this was implemented, there maybe wasn't a `PersistentSettings` thread yet.

To fix that part, `DeletePersistentSettings` should have been changed to use `PostTaskAndReply` with the closure being the reply, so that it would get called from the right thread.

Regardless, I think that the old code could result in a registration modification getting recorded in the map before the map gets wiped by the closure, resulting in the modification being lost, so it seems that this new behavior is better. @sherryzy Do you agree?

Reply:
I think jelle is saying the old implementation had a problem where `registration_map_` was being accessed both in `service_worker_registration_map.cc` by a thread and in `persistent_settings.cc` by another thread (after `DeletePersistentSettings()` runs, it runs the closure).

With the new changes we sidestep that issue entirely since `registration_map_` is accessed immediately and only by the `service_worker_registration_map.cc` thread.

`registration_map_` is entirely separate from the underlying structures of `persistent_settings` so there's no need to tangle them up especially now that the post tasks for persistent_settings are guaranteed to run sequentially and in the same order as the function calls.